vision/classification/resnet/README.md

```diff
@@ -68,24 +68,88 @@ All pre-trained models expect input images normalized in the same way, i.e. mini
+```python
+import onnx
+from onnx import parser
+from onnx import checker
+
+resnet_preproc = parser.parse_model('''
+<
+  ir_version: 8,
+  opset_import: [ "" : 18, "local" : 1 ],
+  metadata_props: [ "preprocessing_fn" : "local.preprocess"]
+>
+resnet_preproc_g (seq(uint8[?, ?, 3]) images) => (float[B, 3, 224, 224] preproc_data)
+{
+    preproc_data = local.preprocess(images)
+}
+
+<
+  opset_import: [ "" : 18 ],
+  domain: "local",
+  doc_string: "Preprocessing function."
+>
+preprocess (input_batch) => (output_tensor) {
+    tmp_seq = SequenceMap <
+        body = sample_preprocessing(uint8[?, ?, 3] sample_in) => (float[3, 224, 224] sample_out) {
+            target_size = Constant <value = int64[2] {256, 256}> ()
+            image_resized = Resize <mode = \"linear\",
+                                    antialias = 1,
+                                    axes = [0, 1],
+                                    keep_aspect_ratio_policy = \"not_smaller\"> (sample_in, , , target_size)
+
+            target_crop = Constant <value = int64[2] {224, 224}> ()
+            image_sliced = CenterCropPad <axes = [0, 1]> (image_resized, target_crop)
+
+            kMean = Constant <value = float[3] {123.675, 116.28, 103.53}> ()
+            kStddev = Constant <value = float[3] {58.395, 57.12, 57.375}> ()
+            im_norm_tmp1 = Cast <to = 1> (image_sliced)
+            im_norm_tmp2 = Sub (im_norm_tmp1, kMean)
+            im_norm = Div (im_norm_tmp2, kStddev)
+
+            sample_out = Transpose <perm = [2, 0, 1]> (im_norm)
+        }
+    > (input_batch)
+    output_tensor = ConcatFromSequence < axis = 0, new_axis = 1 >(tmp_seq)
+}
+
+''')
+checker.check_model(resnet_preproc)
+```
+
+* ResNet preprocessing:
+
+|Model        |Download  |Download (with sample test data)| ONNX version |Opset version|
+|-------------|:--------------|:--------------|:--------------|:--------------|
+|ResNet-preproc| [4.0KB](preproc/resnet-preproc-v1-18.onnx)  |  [864 KB](preproc/resnet-preproc-v1-18.tar.gz) |  1.13.1 | 18|
 
+
+To prepend the data preprocessing steps to the model, we can use the ONNX compose utils:
 ```python
-import numpy
-
-def preprocess(img_data):
-    mean_vec = np.array([0.485, 0.456, 0.406])
-    stddev_vec = np.array([0.229, 0.224, 0.225])
-    norm_img_data = np.zeros(img_data.shape).astype('float32')
-    for i in range(img_data.shape[0]):
-         # for each pixel in each channel, divide the value by 255 to get value between [0, 1] and then normalize
-        norm_img_data[i,:,:] = (img_data[i,:,:]/255 - mean_vec[i]) / stddev_vec[i]
-    return norm_img_data
+
+import onnx
+from onnx import version_converter
+from onnx import checker
+
+network_model = onnx.version_converter.convert_version(network_model, 18)
+network_model.ir_version = 8
+checker.check_model(network_model)
+
+model_w_preproc = onnx.compose.merge_models(
+    preprocessing_model, network_model,
+    io_map=[('preproc_data', 'data')]
+)
+checker.check_model(model_w_preproc)
+
```

Comment:
TODO (separate step). Once https://github.com/onnx/onnx/pull/4976 is merged, add instructions here too